### PR TITLE
Add doc comments to JsonDocument (and friends)

### LINKED ISSUE #35313: Add xml style comments to JsonDocument public APIs
@bartonjs, we need to provide XML comments for the new JsonDocument public types and APIs that you added (similar to the `Utf8JsonReader`/`Ut8JsonWriter`). Would you mind adding them?

https://github.com/dotnet/corefx/pull/34485

### PR DESCRIPTION
**Pull request by bartonjs**
* Add doc comments to all public members of JsonDocument, JsonElement, JsonProperty, and JsonValueType.
* Add "do not use this" doc comments on all internal members, because of source-package concerns.
* Change JsonElement.TokenType from internal to private because it could.

Fixes #35313.

### CODE REVIEW

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs [hunk 1]
```diff
@@ -9,6 +9,9 @@ namespace System.Text.Json
 {
     public sealed partial class JsonDocument
     {
+        /// <summary>
+        ///   This is an implementation detail and MUST NOT be used by source-package consumers.
```
Comment (stephentoub) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs]: Nit: can you remove the extra two-space indent from the beginning of all of the text?

Comment (bartonjs) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs]: I could, I suppose :smile:, but I think it makes it a lot more legible.

Comment (bartonjs) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs]: In particular it's just XML-styled indentation to help identify that a (n e.g. &lt;paramref /&gt;) isn't a new tag, but just still part of the existing tag.

If we have a style thing for it, or there's a tooling problem, I'll certainly go along with it.

Comment (stephentoub) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs]: I just don't remember seeing this style elsewhere in the repo.  Not a big deal, other than I like consistency.  Up to you.

**Comment by danmosemsft** [comment]
```
        /// <summary>
        ///   This is an implementation detail and MUST NOT be called by source-package consumers.
        /// </summary>
```
I would have assumed that source package consumers would not use _any_ `internal` members. Are there some that they are allowed to use? If not, I doubt they would be dissuaded by these comments. I suggest removing them.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 2]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
             CheckSupportedOptions(readerOptions);
 
             return Parse(utf8Json, readerOptions, null);
         }
 
+        /// <summary>
+        ///   Parse a sequence as UTF-8-encoded text representing a single JSON value into a JsonDocument.
+        /// </summary>
+        /// <remarks>
+        ///   The backing <see cref="ReadOnlyMemory{T}"/> values may be used for the entire lifetime of the
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: The `ReadOnlySequence` doesn't necessarily have to be backed by `ReadOnlyMemory`

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 3]
```diff
@@ -71,6 +128,22 @@ public static JsonDocument Parse(Stream utf8Json, JsonReaderOptions readerOption
             }
         }
 
+        /// <summary>
+        ///   Parse a <see cref="Stream"/> as UTF-8-encoded data representing a single JSON value into a
+        ///   JsonDocument.  The Stream will be read to completion.
+        /// </summary>
+        /// <param name="utf8Json">JSON data to parse.</param>
+        /// <param name="readerOptions">Options to control the reader behavior during parsing.</param>
+        /// <param name="cancellationToken">   The token to monitor for cancellation requests.</param>
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: nit: remove extra spacing

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 4]
```diff
@@ -106,6 +179,25 @@ public static JsonDocument Parse(Stream utf8Json, JsonReaderOptions readerOption
             }
         }
 
+        /// <summary>
+        ///   Parse text representing a single JSON value into a JsonDocument.
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: Should we specify the encoding?

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs [hunk 5]
```diff
@@ -14,13 +14,26 @@
 
 namespace System.Text.Json
 {
+    /// <summary>
+    ///   Provides a mechanism for examining the structural content of a JSON value without
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs]: Add something about it being readonly?

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.ObjectEnumerator.cs [hunk 6]
```diff
@@ -25,34 +31,54 @@ internal ObjectEnumerator(JsonElement target)
...
                 _curIdx < 0 ?
                     default :
                     new JsonProperty(new JsonElement(_target._parent, _curIdx));
 
+            /// <summary>
+            ///   Returns an enumerator that iterates the properties of an object.
+            /// </summary>
+            /// <returns>
+            ///   An <see cref="ObjectEnumerator"/> value that can be used to iterate
+            ///   through the array.
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.ObjectEnumerator.cs]: Not an array

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.ObjectEnumerator.cs [hunk 7]
```diff
@@ -25,34 +31,54 @@ internal ObjectEnumerator(JsonElement target)
...
+            /// </returns>
+            /// <remarks>
+            ///   The enumerator will enumerate the properties in the order they are
+            ///   declared, and when an object has multiple definitions of a single
+            ///   property they will all individually be returned (in their correctly
+            ///   defined order).
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.ObjectEnumerator.cs]: `correctly defined order`? Change to something like: `in the order in which they appear in the text`

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 8]
```diff
@@ -7,11 +7,17 @@
 
 namespace System.Text.Json
 {
+    /// <summary>
+    ///   Represents a specific value within a <see cref="JsonDocument"/>.
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: nit: specific JSON value

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 9]
```diff
@@ -23,10 +29,29 @@ internal JsonElement(JsonDocument parent, int idx)
...
 
+        /// <summary>
+        ///   Get the value at a specified index when the current value is a
+        ///   <see cref="JsonValueType.Array"/>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        ///   This value's <see cref="Type"/> is not <see cref="JsonValueType.Array"/>.
+        /// </exception>
+        /// <exception cref="IndexOutOfRangeException">
+        ///   <paramref name="index"/> is not in the range [0, <see cref="GetArrayLength"/>()),
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: remove extra comma (or replace with period)

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 10]
```diff
@@ -85,20 +223,82 @@ public bool TryGetProperty(string propertyName, out JsonElement value)
...
+        ///   is assigned to the value of that property.
+        /// </summary>
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
+        ///   what is matched.
+        /// </remarks>
+        /// <param name="utf8PropertyName">
+        ///   Name of the property to find, encoded as UTF-8 without a Byte-Order-Mark (BOM).
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: Is it worth mentioning the BOM here? Seems unnecessary.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 11]
```diff
@@ -208,13 +593,55 @@ public double GetDouble()
...
         }
 
+        /// <summary>
+        ///   Attempts to represent the current JSON number as a <see cref="float"/>.
+        /// </summary>
+        /// <param name="value">Receives the value.</param>
+        /// <remarks>
+        ///   This method does not parse the contents of a JSON string value.
+        ///
+        ///   On .NET Core this method does not return <c>false</c> for values larger than
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: Should we use `<see langword="true"/>` instead of `<c></c>`?

Comment (ahsonkhan) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: Here and elsewhere we have true/false in the comments.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs [hunk 12]
```diff
@@ -4,17 +4,38 @@
 
 namespace System.Text.Json
 {
+    /// <summary>
+    ///   Represents a single property for a JSON object.
+    /// </summary>
     public readonly struct JsonProperty
     {
+        /// <summary>
+        /// The value of this property
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs]: missing period

File: src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs [hunk 13]
```diff
@@ -4,17 +4,38 @@
...
+        /// The name of this property.
+        /// </summary>
         public string Name => Value.GetPropertyName();
 
+        /// <summary>
+        ///   Provides a <see cref="string"/> representation of the property for
+        ///   debugging purposes.
+        /// </summary>
+        /// <returns>
+        ///   A string containing the un-interpreted value of the property, beginning
```
Comment (ahsonkhan) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonProperty.cs]: `un-interpreted` - try to say that quickly 5x. :)

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs (hunk 1 above)
Comment (ahsonkhan) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.DbRow.cs]: FWIW, this spacing doesn't affect how the comments show up in IntelliSense (though the change does show up in the underlying xml file).

**Comment by ahsonkhan** [comment]
> I would have assumed that source package consumers would not use _any_ `internal` members. Are there some that they are allowed to use?

To avoid having anyone expose the types in their public surface area (since the public APIs are not meant to be exchanged), the `public` types are changed to `internal` within the source package. The consumers don't necessarily know what's public and what's internal, so these comments do provide value here.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs (hunk 4 above)
Comment (ahsonkhan) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: Here/elsewhere we have char overloads.

Comment (bartonjs) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: Since it takes `char` (or `string`) there's no notion of encoding.  Encoding only applies to how raw bytes are interpreted.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs (hunk 11 above)
Comment (bartonjs) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: I changed all of them except the ones in JsonValueType, since they're about the JSON notion of true/false/null, vs the C# one.

**Comment by bartonjs** [comment]
> I would have assumed that source package consumers would not use _any_ `internal` members. Are there some that they are allowed to use? If not, I doubt they would be dissuaded by these comments. I suggest removing them.

No, there are no internal methods that we support being called.  But while writing the docs it occurred to me that someone might notice (e.g.) JsonDocument.GetString and think "yeah, that's what I want... but I can't figure out how to call it" (when they want JsonElement.GetString()).  So this is both a friendly "this is not what you were looking for" and a slightly stronger "we absolutely don't guarantee this method will exist, or behave the same, across different source package versions".

The IntelliSense icon for internal vs public is noticable enough when looking for a difference, but not enough that I think a casual caller would notice they've stepped into the realm of unsupported behavior.

All that said, I don't feel strongly, so if we want to just trust source-package consumers to realize when they've left supported scenarios I'll delete all the internal "MUST NOT" comments.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 14]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
 
+        /// <summary>
+        ///   Parse memory as UTF-8-encoded text representing a single JSON value into a JsonDocument.
+        /// </summary>
+        /// <remarks>
+        ///   The <see cref="ReadOnlyMemory{T}"/> value will be used for the entire lifetime of the
+        ///   JsonDocument object, and the caller must ensure that the data therein does not change during
+        ///   the object lifetime.
+        ///
+        ///   Because the input is considered to be text, a UTF-8 Byte-Order-Mark (BOM) must not be present.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: 📝 This is missing `<para>` elements. Combine the sentences into one paragraph, or wrap both to indicate they are separate.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 15]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
+        /// </remarks>
+        /// <param name="utf8Json">JSON text to parse.</param>
+        /// <param name="readerOptions">Options to control the reader behavior during parsing.</param>
+        /// <returns>
+        ///   A JsonDocument representation of the JSON value.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   A <see cref="JsonDocument"/> representation of the JSON value.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 16]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
+        /// </exception>
         public static JsonDocument Parse(ReadOnlyMemory<byte> utf8Json, JsonReaderOptions readerOptions = default)
         {
             CheckSupportedOptions(readerOptions);
 
             return Parse(utf8Json, readerOptions, null);
         }
 
+        /// <summary>
+        ///   Parse a sequence as UTF-8-encoded text representing a single JSON value into a JsonDocument.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   Parse a sequence as UTF-8-encoded text representing a single JSON value into a <see cref="JsonDocument"/>.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 17]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
+        /// </summary>
+        /// <remarks>
+        ///   The <see cref="ReadOnlySequence{T}"/> may be used for the entire lifetime of the
+        ///   JsonDocument object, and the caller must ensure that the data therein does not change during
+        ///   the object lifetime.
+        ///
+        ///   Because the input is considered to be text, a UTF-8 Byte-Order-Mark (BOM) must not be present.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 18]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
 
             return Parse(utf8Json, readerOptions, null);
         }
 
+        /// <summary>
+        ///   Parse a sequence as UTF-8-encoded text representing a single JSON value into a JsonDocument.
+        /// </summary>
+        /// <remarks>
+        ///   The <see cref="ReadOnlySequence{T}"/> may be used for the entire lifetime of the
+        ///   JsonDocument object, and the caller must ensure that the data therein does not change during
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   <see cref="JsonDocument"/> object, and the caller must ensure that the data therein does not change during
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 19]
```diff
@@ -14,13 +14,55 @@ public sealed partial class JsonDocument
...
+        ///   the object lifetime.
+        ///
+        ///   Because the input is considered to be text, a UTF-8 Byte-Order-Mark (BOM) must not be present.
+        /// </remarks>
+        /// <param name="utf8Json">JSON text to parse.</param>
+        /// <param name="readerOptions">Options to control the reader behavior during parsing.</param>
+        /// <returns>
+        ///   A JsonDocument representation of the JSON value.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   A <see cref="JsonDocument"/> representation of the JSON value.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 20]
```diff
@@ -47,6 +89,21 @@ public static JsonDocument Parse(ReadOnlySequence<byte> utf8Json, JsonReaderOpti
             }
         }
 
+        /// <summary>
+        ///   Parse a <see cref="Stream"/> as UTF-8-encoded data representing a single JSON value into a
+        ///   JsonDocument.  The Stream will be read to completion.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   <see cref="JsonDocument"/>.  The <see cref="Stream"/> will be read to completion.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs [hunk 21]
```diff
@@ -47,6 +89,21 @@ public static JsonDocument Parse(ReadOnlySequence<byte> utf8Json, JsonReaderOpti
...
+        /// </summary>
+        /// <param name="utf8Json">JSON data to parse.</param>
+        /// <param name="readerOptions">Options to control the reader behavior during parsing.</param>
+        /// <returns>
+        ///   A JsonDocument representation of the JSON value.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]:
```suggestion
        ///   A <see cref="JsonDocument"/> representation of the JSON value.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs [hunk 22]
```diff
@@ -14,13 +14,26 @@
...
+    ///   automatically instantiating data values.
+    /// </summary>
+    /// <remarks>
+    ///   This class utilizes resources from pooled memory to minimize the garbage collector (GC)
+    ///   impact in high-usage scenarios. Failure to properly Dispose this object will result in
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs]:
```suggestion
    ///   impact in high-usage scenarios. Failure to properly dispose this object will result in
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.TryGetProperty.cs [hunk 23]
```diff
@@ -9,6 +9,9 @@ namespace System.Text.Json
 {
     public sealed partial class JsonDocument
     {
+        /// <summary>
+        ///   This is an implementation detail and MUST NOT be called by source-package consumers.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.TryGetProperty.cs]: 📝 These comments are not helpful. It would be better to replace them with an attribute that can eventually be supported by a diagnostic analyzer. The comments themselves should be written from the perspective of the caller, which in this case will never be source-package consumers.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs (hunk 21 above)
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: 📝 There are a bunch more cases where `<see cref` is missing around type references.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 24]
```diff
@@ -23,10 +29,29 @@ internal JsonElement(JsonDocument parent, int idx)
...
         public JsonValueType Type => TokenType.ToValueType();
 
+        /// <summary>
+        ///   Get the value at a specified index when the current value is a
+        ///   <see cref="JsonValueType.Array"/>.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">
+        ///   This value's <see cref="Type"/> is not <see cref="JsonValueType.Array"/>.
+        /// </exception>
+        /// <exception cref="IndexOutOfRangeException">
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: ❔  Is this correct? If the arguments are validated, we would typically expect to see `ArgumentOutOfRangeException` for cases less than 0, and `ArgumentException` for cases where it is larger than the length. `IndexOutOfRangeException` is not an `ArgumentException` and is more commonly associated with the `ldelem` and `stelem` IL instructions.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 25]
```diff
@@ -37,13 +62,50 @@ internal JsonElement(JsonDocument parent, int idx)
...
         }
 
+        /// <summary>
+        ///   Gets a <see cref="JsonElement"/> representing the value of a required property identified
+        ///   by <paramref name="propertyName"/>.
+        /// </summary>
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝  Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 26]
```diff
@@ -57,6 +119,30 @@ public JsonElement GetProperty(string propertyName)
...
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 27]
```diff
@@ -67,6 +153,32 @@ public JsonElement GetProperty(ReadOnlySpan<char> propertyName)
...
         }
 
+        /// <summary>
+        ///   Gets a <see cref="JsonElement"/> representing the value of a required property identified
+        ///   by <paramref name="utf8PropertyName"/>.
+        /// </summary>
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 28]
```diff
@@ -67,6 +153,32 @@ public JsonElement GetProperty(ReadOnlySpan<char> propertyName)
...
+        ///   what is matched.
+        /// </remarks>
+        /// <param name="utf8PropertyName">
+        ///   Name of the property whose value to return encoded as UTF-8 with no Byte-Order-Mark (BOM).
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 😕 This sentence is ambiguous: it could be saying the argument is UTF-8 encoded, or saying the return value is UTF-8 encoded. It would help to reword this to eliminate the ambiguity.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 29]
```diff
@@ -77,6 +189,32 @@ public JsonElement GetProperty(ReadOnlySpan<byte> utf8PropertyName)
...
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
+        ///   what is matched.
+        /// </remarks>
+        /// <param name="propertyName">Name of the property to find.</param>
+        /// <param name="value">Receives the value of the located property.</param>
+        /// <returns>
+        ///   <see langword="true"/> if the property was found, <see langword="false"/> otherwise.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]:
```suggestion
        ///   <see langword="true"/> if the property was found; otherwise, <see langword="false"/>.
```

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 30]
```diff
@@ -77,6 +189,32 @@ public JsonElement GetProperty(ReadOnlySpan<byte> utf8PropertyName)
...
 
+        /// <summary>
+        ///   Looks for a property named <paramref name="propertyName"/> in the current object, returning
+        ///   whether or not such a property existed. When the property exists <paramref name="value"/>
+        ///   is assigned to the value of that property.
+        /// </summary>
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 31]
```diff
@@ -85,20 +223,82 @@ public bool TryGetProperty(string propertyName, out JsonElement value)
...
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs (hunk 29 above)
Comment (sharwell) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: This suggestion also applies to the same pattern in a few other places.

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 32]
```diff
@@ -85,20 +223,82 @@ public bool TryGetProperty(string propertyName, out JsonElement value)
...
 
+        /// <summary>
+        ///   Looks for a property named <paramref name="utf8PropertyName"/> in the current object, returning
+        ///   whether or not such a property existed. When the property exists <paramref name="value"/>
+        ///   is assigned to the value of that property.
+        /// </summary>
+        /// <remarks>
+        ///   Property name matching is performed as an ordinal, case-sensitive, comparison.
+        ///
+        ///   If a property is defined multiple times for the same object, the last such definition is
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 33]
```diff
@@ -191,13 +534,55 @@ public ulong GetUInt64()
...
         }
 
+        /// <summary>
+        ///   Gets the current JSON number as a <see cref="double"/>.
+        /// </summary>
+        /// <returns>The current JSON number as a <see cref="double"/>.</returns>
+        /// <remarks>
+        ///   This method does not parse the contents of a JSON string value.
+        ///
+        ///   On .NET Core this method returns <see cref="double.PositiveInfinity"/> (or
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 34]
```diff
@@ -191,13 +534,55 @@ public ulong GetUInt64()
...
         }
 
+        /// <summary>
+        ///   Attempts to represent the current JSON number as a <see cref="double"/>.
+        /// </summary>
+        /// <param name="value">Receives the value.</param>
+        /// <remarks>
+        ///   This method does not parse the contents of a JSON string value.
+        ///
+        ///   On .NET Core this method does not return <see langword="false"/> for values larger than
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 35]
```diff
@@ -208,13 +593,55 @@ public double GetDouble()
...
         }
 
+        /// <summary>
+        ///   Attempts to represent the current JSON number as a <see cref="float"/>.
+        /// </summary>
+        /// <param name="value">Receives the value.</param>
+        /// <remarks>
+        ///   This method does not parse the contents of a JSON string value.
+        ///
+        ///   On .NET Core this method does not return <see langword="false"/> for values larger than
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 💡 Consider rewording this to match `GetDouble()`

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 36]
```diff
@@ -208,13 +593,55 @@ public double GetDouble()
...
         }
 
+        /// <summary>
+        ///   Gets the current JSON number as a <see cref="float"/>.
+        /// </summary>
+        /// <returns>The current JSON number as a <see cref="float"/>.</returns>
+        /// <remarks>
+        ///   This method does not parse the contents of a JSON string value.
+        ///
+        ///   On .NET Core this method returns <see cref="float.PositiveInfinity"/> (or
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs [hunk 37]
```diff
@@ -291,6 +792,26 @@ public ObjectEnumerator EnumerateObject()
             return new ObjectEnumerator(this);
         }
 
+        /// <summary>
+        ///   Gets a string representation for the current value appropriate to the value type.
+        /// </summary>
+        /// <remarks>
+        ///   For <see cref="JsonValueType.Null"/>, <see cref="string.Empty"/> is returned.
+        ///
+        ///   For <see cref="JsonValueType.True"/>, <see cref="bool.TrueString"/> is returned.
```
Comment (sharwell) [inline src/System.Text.Json/src/System/Text/Json/Document/JsonElement.cs]: 📝 Missing `<para>` elements

File: src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs (hunk 21 above)
Comment (bartonjs) [reply src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs]: They're following what I _think_ the style guide is, which is the first usage in a given section is a link and the others are not.  And I didn't see=thistype, since that felt pretty weird.
